src/arrayMethodJoin.js

```diff
@@ -5,7 +5,29 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    if (this.length === 0) {
+      return '';
+    }
+
+    let finalSeparator = separator;
+
+    if (separator === undefined) {
+      finalSeparator = ',';
+    }
```

Comment:
```suggestion
```
looks like redundant.
instead to check separator on undefined set `','` as default argument.